packages/plugin-ext/src/main/browser/view/tree-view-widget.tsx

```diff
@@ -254,7 +254,7 @@ export class TreeViewWidget extends TreeWidget {
         return React.createElement('div', attrs, ...children);
     }
 
-    protected getCaption(node: TreeNode): React.ReactNode[] {
+    protected getCaption(node: TreeNode): React.ReactNode {
```

Comment:
The main trick is that we turn each element to span that it does not create multiple lines and then wrap them into div element with block display and `noWrapInfo` class install. It makes sure that all span are treated as one element and get trimmed accordingly only once.

Reply:
I had to solve this issue already many times and each time it is hard for me, had to remember that text overflow does not play nicely with flex display.